airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/io/IcebergUtil.kt

```diff
@@ -180,9 +188,36 @@ object IcebergUtil {
             }
     }
 
+    fun toIcebergSchema(stream: DestinationStream, pipeline: MapperPipeline): Schema {
+        val primaryKeys =
+            when (stream.importType) {
+                is Dedupe -> (stream.importType as Dedupe).primaryKey
+                else -> emptyList()
+            }
+        return pipeline.finalSchema.withAirbyteMeta(true).toIcebergSchema(primaryKeys)
+    }
+
     private fun getSortOrder(schema: Schema): SortOrder {
         val builder = SortOrder.builderFor(schema)
         schema.identifierFieldNames().forEach { builder.asc(it) }
         return builder.build()
     }
+
+    private fun getOperation(
+        @Suppress("UNUSED_PARAMETER") record: DestinationRecord,
+        importType: ImportType,
+    ): Operation =
+        // TODO This disabled out of caution due to the potential cost of serializing every record
+        // to check
+        //        the deletion status.  This should be revisited when there is a cheaper way to do
+        // this, such
+        //        as after a protocol change that explicitly states the operation.
+        //        if (record.data.toJson().get(AIRBYTE_CDC_DELETE_COLUMN) != null) {
```

Comment:
can we do something like this to dodge the serialize?
```
(record.data as ObjectValue).properties[DELETE_COLUMN] != null
```

Reply:
@edgao Sure?  I'll test it out and update the PR if it works

Reply:
@edgao That appeared to work.  I will update the PR to include the DELETE operation detection based on your suggestion.